yt/fields/field_info_container.py

```diff
@@ -402,6 +404,12 @@ def add_output_field(self, name, sampling_type, **kwargs):
         kwargs.setdefault("ds", self.ds)
         self[name] = DerivedField(name, sampling_type, NullFunc, **kwargs)
 
+    def __setitem__(self, key, value):
+        ftype, fname = key
+        if any(fname == _fname for _ftype, _fname in self):
+            self._ambiguous_field_names[fname].append(ftype)
```

Comment:
I'd be happier with an implementation that couldn't append redundant values, though I could easily be convinced that this is fine.

Reply:
I've changed it to a set!

Reply:
Perfect. This was my original suggestion but I forgot about `set.add` and thought there wasn't an equivalent to `list.append` :)